src/main/java/our/member/service/UserAuthenticationService.java

```diff
@@ -0,0 +1,63 @@
+package our.member.service;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+import our.member.domain.TokenEntity;
+import our.member.domain.UserEntity;
+import our.member.repository.TokenRepository;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class UserAuthenticationService implements authenticationService{
+    private final TokenRepository tokenRepository;
+    private static Map<String, String> mailSender = new HashMap<>();
+
+    public UserAuthenticationService(TokenRepository tokenRepository) {
+        this.tokenRepository = tokenRepository;
+    }
+
+    @Override
+    public void sendTokenToEmail(UserEntity userEntity) {
+        // token 생성
+        String token = userEntity.getEmail() + userEntity.getUsername();
+        TokenEntity tokenEntity = new TokenEntity();
+        tokenEntity.setToken(token);
+        tokenEntity.setUserId(userEntity.getUserId());
+        tokenRepository.save(tokenEntity);
+
+        mailSender.put("email", userEntity.getEmail());
+        mailSender.put("title", "인증 이메일");
+        mailSender.put("message", "인증 토큰: "+token);
+
+        for (String key : mailSender.keySet()) {
+            System.out.println(key + ": " + mailSender.get(key));
+        }
+
+
+    }
+
+    @Override
+    public boolean verifyToken(UserEntity userEntity, String token) {
+        Optional<TokenEntity> tokenOptional = tokenRepository.findByToken(token);
```

Comment:
이렇게 진행하면 아래 부분도 다 해결할 수 있을 것 같아요!
```suggestion
        Optional<TokenEntity> tokenOptional = tokenRepository.findByToken(token).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
```